src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClientDiscovery.cs

```diff
@@ -49,6 +50,7 @@ private sealed record ResolveResult(Contact Contact, IActorRef? Subject);
     private readonly string _targetActorSystemName;
     private readonly string _receptionistName;
     private readonly string _transportProtocol;
+    private readonly int _numberOfContacts;
```

Comment:
This is just the number of initial contacts we need, right?

Reply:
it limits the number of contacts we use when we instantiate the ClusterClient actor. Too many of these will set the initial contacts to a higher than optimum number. The thing is that ClusterClient initial contacts never gets trimmed when they can't be contacted, if we use the whole discovery resolve result as the ClusterClient initial contact list, there will be a big burst of network connection attempt when the ClusterClient actor starts up and whenever its internal contact list could not be contacted anymore.

Reply:
Got it, makes sense